src/Disks/ObjectStorages/Cached/registerDiskCache.cpp

```diff
@@ -49,7 +51,40 @@ void registerDiskCache(DiskFactory & factory, bool /* global_skip_access_check *
             file_cache_settings.loadFromConfig(config, config_prefix);
 
         auto config_fs_caches_dir = context->getFilesystemCachesPath();
-        if (config_fs_caches_dir.empty())
+        if (custom_disk)
+        {
+            static constexpr auto custom_cached_disks_base_dir_in_config = "custom_cached_disks_base_directory";
+            auto custom_cached_disk_path_prefix = context->getConfigRef().getString(custom_cached_disks_base_dir_in_config, config_fs_caches_dir);
+            if (custom_cached_disk_path_prefix.empty())
+            {
+                if (!attach)
+                {
+                    throw Exception(
```

Comment:
How it works if only `filesystem_caches_path` is set? We will check that `custom_cached_disk_path_prefix.empty()` and `!attach` and throw the exception

Reply:
See `auto custom_cached_disk_path_prefix = context->getConfigRef().getString(custom_cached_disks_base_dir_in_config, config_fs_caches_dir);` here `config_fs_caches_dir` is the default and it is equal to `filesystem_caches_path` from config.